---
slug: /api/misc/@docusaurus/eslint-plugin/no-html-links
---

# no-html-links

import APITable from '@site/src/components/APITable';

Ensure that the Docusaurus [`<Link>`](../../../docusaurus-core.mdx#link) component is used instead of `<a>` tags.

The `<Link>` component has prefetching and preloading built-in. 또한 빌드 시 끊어진 링크를 감지하고 도큐사우루스가 사이트 구조를 더 잘 이해할 수 있게 도와줍니다.

## Rule Details {#details}

Examples of **incorrect** code for this rule:

```html
<a href="/page">go to page!</a>

<a href="https://x.com/docusaurus" target="_blank">X</a>
```

Examples of **correct** code for this rule:

```js
import Link from '@docusaurus/Link'

<Link to="/page">go to page!</Link>

<Link to="https://x.com/docusaurus">X</Link>
```

## Rule Configuration {#configuration}

설정할 수 있는 필드

```mdx-code-block
<APITable>
```

| 옵션                    | 타입        | 기본값     | 설명                                                                                  |
| --------------------- | --------- | ------- | ----------------------------------------------------------------------------------- |
| `ignoreFullyResolved` | `boolean` | `false` | Set to true will not report any `<a>` tags with absolute URLs including a protocol. |

```mdx-code-block
</APITable>
```
