diff --git a/lib/index.js b/lib/index.js
index 1167aa26f6ad3cbacde07230386a54270e0bcdb9..a6069a0acfbac9003b68232f9e471663d54cfe03 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -159,6 +159,9 @@ function createLoadableComponent(loadFn, options) {
         loading: res.loading,
         loaded: res.loaded
       };
+
+      this._loadModule();
+
       return _this;
     }
 
@@ -168,10 +171,6 @@ function createLoadableComponent(loadFn, options) {
 
     var _proto = LoadableComponent.prototype;
 
-    _proto.UNSAFE_componentWillMount = function UNSAFE_componentWillMount() {
-      this._loadModule();
-    };
-
     _proto.componentDidMount = function componentDidMount() {
       this._mounted = true;
     };
