diff --git a/node_modules/react-loadable-ssr-addon/lib/ReactLoadableSSRAddon.js b/node_modules/react-loadable-ssr-addon/lib/ReactLoadableSSRAddon.js
index 94b0076..447e587 100644
--- a/node_modules/react-loadable-ssr-addon/lib/ReactLoadableSSRAddon.js
+++ b/node_modules/react-loadable-ssr-addon/lib/ReactLoadableSSRAddon.js
@@ -81,19 +81,9 @@ var ReactLoadableSSRAddon = function () {
         modules = _ref.modules;
     var origins = new Set();
 
-    for (var i = 0; i < modules.length; i += 1) {
-      var reasons = modules[i].reasons;
-
-      for (var j = 0; j < reasons.length; j += 1) {
-        var reason = reasons[j];
-        var type = reason.dependency ? reason.dependency.type : null;
-        var userRequest = reason.dependency ? reason.dependency.userRequest : null;
-
-        if (type === 'import()') {
-          origins.add(userRequest);
-        }
-      }
-    }
+    // for (var i = 0; i < modules.length; i += 1) {
+    //   console.log(modules[i]);
+    // }
 
     if (origins.size === 0) {
       return [names[0] || id];
