---
slug: /api/misc/@docusaurus/eslint-plugin/no-untranslated-text
---

# no-untranslated-text

import APITable from '@site/src/components/APITable';

确保 JSX 中的所有文本标签都被翻译 API 包裹。

When the [i18n feature](../../../i18n/i18n-introduction.mdx) is used, this rule ensures that all labels appearing on the website are translatable, so no string accidentally slips through untranslated.

## Rule Details {#details}

Examples of **incorrect** code for this rule:

```js
// Hello World is not translated
<Component>Hello World</Component>
```

Examples of **correct** code for this rule:

```js
// Hello World is translated
<Component>
  <Translate>Hello World</Translate>
</Component>
```

## Rule Configuration {#configuration}

接受的字段：

```mdx-code-block
<APITable>
```

| 选项               | 类型         | 默认值  | 描述                           |
| ---------------- | ---------- | ---- | ---------------------------- |
| `ignoredStrings` | `string[]` | `[]` | 如果文本标签完全由此列表包含的字符串组成，将不会被报告。 |

```mdx-code-block
</APITable>
```

## When Not To Use It {#when-not-to-use}

If you're not using the [i18n feature](../../../i18n/i18n-introduction.mdx), you can disable this rule. 你也可以在不应被翻译的文本处禁用这条规则。

## Further Reading {#further-reading}

- https://docusaurus.io/zh-CN/docs/docusaurus-core#translate
- https://docusaurus.io/zh-CN/docs/docusaurus-core#translate-imperative
